node-red-contrib-ms-language/node-luis.js

```diff
@@ -1,6 +1,113 @@
+                value: entity.text,
+                children: []
+            });
+            Object.values(prediction.entities[entity.type][0].$instance).forEach(child => {
+                compositeEntities[compositeEntities.length - 1].children.push({
+                  type: child[0].type,
+                  value: child[0].text
+                });
+                entities.push({
+                  type: child[0].type,
+                  entity: child[0].text,
+                  startIndex: child[0].startIndex,
+                  endIndex: child[0].startIndex + child[0].length,
+                  score: entity.score
+                })
+              });
+        } else {
+            entities.push({
+                type: entity.type,
+                entity: entity.text,
+                startIndex: entity.startIndex,
+                endIndex: entity.startIndex + entity.length, // add end index
+                resolution: {
+                  values: response.prediction.entities[entity.type.replace('builtin.', '')].reduce((result, e) => {
+                    if (Array.isArray(e)) {
+                      return result.concat(...e);
+                    }
+                    result.push(e);
+                    return result;
+                  }, [])
+                }
+            });
+        }
+    });
+
+    return {
+        entities,
+        compositeEntities
+    };
+};
+
+const asyncGetPrediction = async function(node, text) {
+    let entityObject = {
+        entities: [],
+        compositeEntities: []
+    };
+
+    const response = await request({
+        uri: `${node.endpoint}${encodeURIComponent(text)}`,
+        method: 'GET',
+        headers: {
+            'Ocp-Apim-Subscription-Key': node.subKey,
+            ...(node.spellCheckKey && {'mkt-bing-spell-check-key': node.spellCheckKey})
+        },
+        json: true
+    });
+
+    // Merge additional metadate in the $instance object to entities
+    if (response.prediction.entities.$instance) {
+        entityObject = extractEntities(response.prediction);
+    }
+
+    return {
+        query: response.query,
+        alteredQuery: response.prediction.alteredQuery,
+        intents: response.prediction.intents,
+        entities: entityObject.entities,
+        compositeEntities: entityObject.compositeEntities,
+        topScoringIntent: {
+            intent: response.prediction.topIntent,
+            score: response.prediction.intents[response.prediction.topIntent].score
+        },
+        source: 'luis'
+    };
+}
+
+
+const input = function(RED, node, data, config) {
+    // Log activity
+    try { setTimeout(function() { helper.trackActivities(node)},0); }
+    catch(err) { console.log(err); }
+
+    // Get parameters
+    let output = config.intent || 'payload';
+    let text = helper.getContextValue(RED, node, data, config.text || 'payload', config.textType);
+
+    asyncGetPrediction(node, text).then(response => {
```

Comment:
you can change "input" to an async function and use await with a try/catch to get the result of asyncGetPrediction instead of a promise. Moreover, I think you can just name the function getPrediction.